assets/app/scripts/controllers/catalog/images.js

```diff
@@ -35,6 +34,8 @@ angular.module('openshiftConsole')
                 }
                 if (categoryTags.indexOf("builder") >= 0) {
                   $scope.builders.push(image);
+                } else {
```

Comment:
this function feels buggy... it's sharing `categoryTags` between loops and potentially adding the same image to the builders/nonBuilders lists multiple times